src/databricks/labs/remorph/reconcile/query_builder.py

```diff
@@ -8,129 +9,93 @@
+        return transformation_rule_mapping
 
+    def _get_default_transformation(self, columns, column_mapping, schema):
         transformation_rule_mapping = []
         for column in columns:
-            if column_mapping_dict and self.layer == "target":
-                transform_column = (
-                    column_mapping_dict.get(column).source_name if column_mapping_dict.get(column) else column
-                )
-            else:
-                transform_column = column
-
-            if transformations_dict and transform_column in transformations_dict.keys():
-                transformation = self._get_layer_transform(transformations_dict, transform_column, self.layer)
-            else:
-                column_data_type = schema.get(column).data_type
-                transformation = self._get_default_transformation(self.source, column_data_type).format(column)
+            column_origin = column if self.qc.layer == "source" else self._get_column_map(column, column_mapping)
+            column_data_type = schema.get(column_origin).data_type
+            transformation = self._get_default_transformation_expr(self.qc.db_type, column_data_type).format(
+                column_origin
+            )
 
-            if column_mapping_dict and column in column_mapping_dict.keys():
-                column_alias = column_mapping_dict.get(column).source_name
-            else:
-                column_alias = column
+            column_origin, column_alias = self._get_column_alias(self.qc.layer, column, column_mapping)
 
-            transformation_rule_mapping.append(TransformRuleMapping(column, transformation, column_alias))
+            transformation_rule_mapping.append(TransformRuleMapping(column_origin, transformation, column_alias))
 
         return transformation_rule_mapping
 
     @staticmethod
-    def _get_default_transformation(data_source: str, data_type: str) -> str:
-        if data_source == "oracle":
+    def _get_default_transformation_expr(data_source: str, data_type: str) -> str:
+        if data_source == SourceType.ORACLE.value:
             return OracleDataSource.oracle_datatype_mapper.get(data_type, ColumnTransformationType.ORACLE_DEFAULT.value)
-        if data_source == "snowflake":
+        if data_source == SourceType.SNOWFLAKE.value:
             return SnowflakeDataSource.snowflake_datatype_mapper.get(
                 data_type, ColumnTransformationType.SNOWFLAKE_DEFAULT.value
             )
-        if data_source == "databricks":
+        if data_source == SourceType.DATABRICKS.value:
             return DatabricksDataSource.databricks_datatype_mapper.get(
                 data_type, ColumnTransformationType.DATABRICKS_DEFAULT.value
             )
         msg = f"Unsupported source type --> {data_source}"
         raise ValueError(msg)
 
+    def _generate_transformation_rule_mapping(self, columns: list[str]) -> list[TransformRuleMapping]:
```

Comment:
It looks like this method can be reused up in the code

Reply:
actually _generate_transformation_rule_mapping () : # calls below two functions
                           _get_custom_transformation() # get the user defined transformations
                           _get_default_transformation() # get the default transformations